File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(string n1, string n2) {
    return n1 + n2 > n2 + n1;
}

string solution(vector<int> numbers) {
    string ans = "";
    vector<string> snum;
    
    for(auto n : numbers) snum.push_back(to_string(n));
    
    sort(snum.begin(), snum.end(), compare);
    
    if(snum[0] == "0") return "0";
    
    for(auto n : snum) ans += n;
    return ans;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        StringBuilder ans = new StringBuilder();
        
        String[] nums = Arrays.stream(numbers)
            .mapToObj(String::valueOf)
            .toArray(String[]::new);
        
        Arrays.sort(nums, (n1, n2) -> (n2 + n1).compareTo(n1 + n2));
        
        if(nums[0].equals("0")) return "0";
        
        Arrays.stream(nums).forEach(n -> {
            ans.append(n);
        });
        
        return ans.toString();
    }
}